File: fix_duplicate_exports.py
#!/usr/bin/env python3
"""
Script to fix duplicate export statements in TypeScript/JSX files
"""
import os
import re
import glob

def fix_duplicate_exports(file_path):
    """Fix duplicate export statements in a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Find all export default statements
        export_matches = list(re.finditer(r'export default \w+;', content))
        
        if len(export_matches) > 1:
            # Keep only the last export statement
            for i, match in enumerate(export_matches[:-1]):
                content = content.replace(match.group(), '')
        
        # Remove any remaining PagePage references
        content = re.sub(r'export default PagePage;', '', content)
        
        # Clean up multiple empty lines
        content = re.sub(r'\n\s*\n\s*\n', '\n\n', content)
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed duplicate exports in: {file_path}")
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix duplicate exports in all files"""
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'components/**/*.tsx',
        'components/**/*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if os.path.isfile(file_path):
                files_processed += 1
                if fix_duplicate_exports(file_path):
                    files_fixed += 1
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed duplicate exports in {files_fixed} files")

if __name__ == "__main__":
    main()

File: fix_duplicate_exports.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

function findFilesWithDuplicateExports(dir) {
  const files = [];
  
  function searchDirectory(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
        searchDirectory(fullPath);
      } else if (item.endsWith('.tsx') || item.endsWith('.ts')) {
        try {
          const content = fs.readFileSync(fullPath, 'utf8');
          const exportMatches = content.match(/export default/g);
          
          if (exportMatches && exportMatches.length > 1) {
            files.push(fullPath);
          }
        } catch (error) {
          // Skip files that can't be read
        }
      }
    }
  }
  
  searchDirectory(dir);
  return files;
}

function fixDuplicateExports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Find all export default statements
    const lines = content.split('\n');
    const exportLines = [];
    
    for (let i = 0; i < lines.length; i++) {
      if (lines[i].trim().startsWith('export default')) {
        exportLines.push({ line: i, content: lines[i] });
      }
    }
    
    if (exportLines.length > 1) {
      console.log(`Fixing ${filePath} - found ${exportLines.length} export default statements`);
      
      // Keep only the last export default statement
      const lastExport = exportLines[exportLines.length - 1];
      
      // Remove all export default statements
      for (let i = exportLines.length - 1; i >= 0; i--) {
        const exportLine = exportLines[i];
        lines.splice(exportLine.line, 1);
      }
      
      // Add the last export default at the end
      lines.push('');
      lines.push(lastExport.content);
      
      const newContent = lines.join('\n');
      fs.writeFileSync(filePath, newContent);
      
      console.log(`✅ Fixed ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.log(`❌ Error fixing ${filePath}: ${error.message}`);
    return false;
  }
}

function main() {
  console.log('🔍 Searching for files with duplicate export default statements...');
  
  const files = findFilesWithDuplicateExports('./app');
  console.log(`Found ${files.length} files with duplicate exports`);
  
  let fixedCount = 0;
  
  for (const file of files) {
    if (fixDuplicateExports(file)) {
      fixedCount++;
    }
  }
  
  console.log(`\n📊 Summary:`);
  console.log(`✅ Fixed: ${fixedCount} files`);
  console.log(`❌ Failed: ${files.length - fixedCount} files`);
  
  console.log('\n🎉 Duplicate export fix completed!');
}

main();